### PR TITLE
minor import edits and use of _illegal

### LINKED ISSUE #23153: importing sympy gives error in current master
https://github.com/sympy/sympy/pull/22677 seems to have caused this @smichr 

on current master locally:
```python
>>> import sympy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/c/Users/paul/Documents/PythonScripts/Libs/SymPy/sympy/sympy/__init__.py", line 151, in <module>
    from .concrete import product, Product, summation, Sum
  File "/mnt/c/Users/paul/Documents/PythonScripts/Libs/SymPy/sympy/sympy/concrete/__init__.py", line 1, in <module>
    from .products import product, Product
  File "/mnt/c/Users/paul/Documents/PythonScripts/Libs/SymPy/sympy/sympy/concrete/products.py", line 4, in <module>
    from .summations import Sum, summation, _dummy_with_inherited_properties_concrete
  File "/mnt/c/Users/paul/Documents/PythonScripts/Libs/SymPy/sympy/sympy/concrete/summations.py", line 31, in <module>
    from sympy.polys.polyutils import illegal
ImportError: cannot import name 'illegal' from 'sympy.polys.polyutils' (/mnt/c/Users/paul/Documents/PythonScripts/Libs/SymPy/sympy/sympy/polys/polyutils.py)
```
It fails the code quality check in the CI too.

### PR DESCRIPTION
**Pull request by smichr**
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/concrete/summations.py [hunk 1]
```diff
@@ -1617,3 +1617,6 @@ def _dummy_with_inherited_properties_concrete(limits):
     if assumptions_to_add:
         assumptions_to_keep.update(assumptions_to_add)
         return Dummy('d', **assumptions_to_keep)
+
+    _illegal = (S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity)
+
```
Comment (smichr) [inline sympy/concrete/summations.py:1622]:
```suggestion
_illegal = (S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity)
```

File: sympy/polys/domains/tests/test_domains.py [hunk 2]
```diff
@@ -819,8 +819,8 @@ def test_RealField_from_sympy():
 
 
 def test_not_in_any_domain():
-    check = list(_illegal) + [x] + [
-        float(i) for i in _illegal if i != S.ComplexInfinity]
+    check = _illegal + [x] + [
```
Comment (smichr) [inline sympy/polys/domains/tests/test_domains.py:822]:
```suggestion
    check = list(_illegal) + [x] + [
```

**Comment by smichr** [comment]
will squash and commit asap

**Comment by oscarbenjamin** [comment]
Does this need to be more complicated than #23155?

**Comment by smichr** [comment]
> Does this need to be more complicated than #23155?

I was doing the best I could from a branch that was older (I guess). If the tests pass on yours, the answer is "no".

**Pull request merged**